libevmasm/Assembly.cpp

```diff
@@ -1281,163 +1329,119 @@ LinkerObject const& Assembly::assembleLegacy() const
-				item.setPushedValue(u256(s));
-				unsigned b = std::max<unsigned>(1, numberEncodingSize(s));
-				ret.bytecode.push_back(static_cast<uint8_t>(pushInstruction(b)));
-				ret.bytecode.resize(ret.bytecode.size() + b);
-				bytesRef byr(&ret.bytecode.back() + 1 - b, b);
-				toBigEndian(s, byr);
-			});
+			assertThrow(item.data() <= std::numeric_limits<size_t>::max(), AssemblyException, "");
+			auto s = subAssemblyById(static_cast<size_t>(item.data()))->assemble().bytecode.size();
+			item.setPushedValue(u256(s));
+			unsigned b = std::max<unsigned>(1, numberEncodingSize(s));
+			ret.bytecode.push_back(static_cast<uint8_t>(pushInstruction(b)));
+			ret.bytecode.resize(ret.bytecode.size() + b);
+			bytesRef byr(&ret.bytecode.back() + 1 - b, b);
+			toBigEndian(s, byr);
 			break;
 		}
 		case PushProgramSize:
-		{
-			assembleInstruction([&](){
-				ret.bytecode.push_back(dataRefPush);
-				sizeRefs.push_back(static_cast<unsigned>(ret.bytecode.size()));
-				ret.bytecode.resize(ret.bytecode.size() + bytesPerDataRef);
-			});
+			ret.bytecode.push_back(dataRefPush);
+			sizeRefs.push_back(static_cast<unsigned>(ret.bytecode.size()));
+			ret.bytecode.resize(ret.bytecode.size() + bytesPerDataRef);
 			break;
-		}
 		case PushLibraryAddress:
 		{
-			assembleInstruction([&]() {
-				auto const [bytecode, linkRef] = assemblePushLibraryAddress(item, ret.bytecode.size());
-				ret.bytecode += bytecode;
-				ret.linkReferences.insert(linkRef);
-			});
+			auto const [bytecode, linkRef] = assemblePushLibraryAddress(item, ret.bytecode.size());
+			ret.bytecode += bytecode;
+			ret.linkReferences.insert(linkRef);
 			break;
 		}
 		case PushImmutable:
-			assembleInstruction([&]() {
-				ret.bytecode.push_back(static_cast<uint8_t>(Instruction::PUSH32));
-				// Maps keccak back to the "identifier" std::string of that immutable.
-				ret.immutableReferences[item.data()].first = m_immutables.at(item.data());
-				// Record the bytecode offset of the PUSH32 argument.
-				ret.immutableReferences[item.data()].second.emplace_back(ret.bytecode.size());
-				// Advance bytecode by 32 bytes (default initialized).
-				ret.bytecode.resize(ret.bytecode.size() + 32);
-			});
+			ret.bytecode.push_back(static_cast<uint8_t>(Instruction::PUSH32));
+			// Maps keccak back to the "identifier" std::string of that immutable.
+			ret.immutableReferences[item.data()].first = m_immutables.at(item.data());
+			// Record the bytecode offset of the PUSH32 argument.
+			ret.immutableReferences[item.data()].second.emplace_back(ret.bytecode.size());
+			// Advance bytecode by 32 bytes (default initialized).
+			ret.bytecode.resize(ret.bytecode.size() + 32);
 			break;
 		case VerbatimBytecode:
 			ret.bytecode += assembleVerbatimBytecode(item);
 			break;
 		case AssignImmutable:
 		{
+			// this decomposes into multiple evm instructions, so we manually call emit
 			// Expect 2 elements on stack (source, dest_base)
```

Comment:
This looks like a single, confusing comment, especially without consistent capitalization and spacing.

```suggestion
			// This item type decomposes into multiple evm instructions, so we manually call emit()

			// Expect 2 elements on stack (source, dest_base)
```

It would probably be clearer to just put it next to the first `emit()`.

Reply:
Fair!